From a557e6261afc87d3849b557257185ff72be15a82 Mon Sep 17 00:00:00 2001
From: Ian Kent <raven@themaw.net>
Date: Sun, 31 May 2020 08:47:29 +0800
Subject: [PATCH 10/13] autofs-5.1.6 - fix double quoting in auto.smb

The example program mount script installed to /etc/auto.smb incorrectly
adds a quote for the trailing dollar of special Windows mounts. But they
are already surrounded by double quotes. This may have been handled by
mount.cifs at some point but it's failing now.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 samples/auto.smb | 2 --
 1 file changed, 2 deletions(-)

diff --git a/samples/auto.smb b/samples/auto.smb
index 6af5d85..4842b68 100755
--- a/samples/auto.smb
+++ b/samples/auto.smb
@@ -75,8 +75,6 @@ $SMBCLIENT $smbopts -gL "$key" 2>/dev/null| awk -v "key=$key" -v "opts=$opts" -F
 		  dir = $2
 		  loc = $2
 		  # Enclose mount dir and location in quotes
-		  # Double quote "$" in location as it is special
-		  gsub(/\$$/, "\\$", loc);
 		  gsub(/\&/,"\\\\&",loc)
 		  print " \\\n\t \"/" dir "\"", "\"://" key "/" loc "\""
 		}
-- 
2.27.0

